src/media/js/apps.js

```diff
@@ -117,6 +117,16 @@ define('apps',
             (!capabilities.packagedWebApps && product.is_packaged) ||
             !_.contains(product.device_types, device)) {
             reasons.push(gettext('not available for your platform'));
+        } else if (product.feature_compatibility === false) {
+            // If feature_compatibility is false (and not just null!), then it
+            // means the app requires features we don't have.
+            reasons.push(gettext('not compatible with your device'));
+        } else if (product.isAddon && !settings.addonsEnabled) {
+            // If we're dealing with an Addon but the setting is false, it
+            // means we're using an older version of Firefox OS or that we are
+            // in the browser and can't detect compatibility.
+            // FIXME: find the right text to use, show a banner ?
+            reasons.push(gettext('not compatible with your FxOS version'));
```

Comment:
I'm changing the string to the generic 'not compatible with your device' for now; it's not displayed on add-ons tiles anyway atm.